run_py_tools.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+CMDS=("black --check --diff sml_small tests" "flake8 sml_small tests" "isort --check-only ." "bandit -c pyproject.toml -r .")
```

Comment:
My understanding is this file is built from the sml-catalogue file so it may just be we are copying the existing functionality but do you know the reasoning for running black etc as check only? seems weird to me that we use the tool to check if everything has been run rather than just having the tool run everything for us

Reply:
This is copying the functionality from the catalogue, I believe having the tools automatically correct any formatting errors might cause some confusion since the corrections would not automatically be committed. I think it's clearer to have the tools in check-only mode, and then you can manually run black etc. if needed. happy to look into it more if making the changes automatically is better

Reply:
I think from my perspective the way I see this working is that the CI exists to ensure formatting tools have been run and because it does this we've essentially made the decision that whatever black, isort etc. does to the files is "correct" and that the tool makes more sense to be used as a one stop shop to run everything we need to rather than it being used to check if we have.

@jhmjam @gibbardsteve any thoughts on this?

Reply:
I had a chat with @JasonBellONS and I am happy with either approach but I do think Jason's approach is nicer. For everyone's information, the context behind this is that originally when I built the shell file for the sml catalogue repo, I kept the commands the same/ consistent with the commands in the pipeline files. Now @joepercival carried that on here to keep it in line.

But I am happy with either approach to check only or making the changes automatically as I don't see a big problem. I thought the reason maybe check only was used in the first place by Phil Bambridge's team was maybe because they didn't want automatic code changes in the pipelines like formatting and or in case they needed to make exceptions.

I am happy with any approach that's best =)

Reply:
I'll take a look into it, the main difficulty would be how to automatically commit any changes made by black etc.

Reply:
Do you mean before you push locally to the repo or on the pipeline?

Reply:
I mean pushing to the repo, if we have black etc. making changes pre-push, then we need to commit those changes, I think I have a solution